src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/jpa/repository/CaseAllocationRepository.kt

```diff
@@ -46,4 +46,11 @@ interface CaseAllocationRepository : JpaRepository<CaseAllocationEntity, Long> {
       " AND A.isDeleted=false",
   )
   fun findTotalCaseCountByPrisonId(prisonId: String): Int
+
+  @Query(
```

Comment:
It might be better to get just one record from the database (the one with the latest `when_created` value) rather than just assuming this query can only return one record (it will error out if multiple records come back).

Reply:
Done